src/services/database/dynamo-ehr-transfer-tracker.js

```diff
@@ -0,0 +1,170 @@
+      throw new Error('The given argument `updateParams` is not an array');
+    }
+    const command = new TransactWriteCommand({
+      TransactItems: updateParams.map((params) => ({
+        Update: {
+          TableName: this.tableName,
+          ...params,
+        },
+      })),
+    });
+
+    await this.client.send(command);
+  }
+
+  async queryTableByNhsNumber(nhsNumber, includeDeletedRecord = false) {
+    const params = {
+      TableName: this.tableName,
+      IndexName: 'NhsNumberSecondaryIndex',
+      ExpressionAttributeValues: {
+        ':nhsNumber': nhsNumber,
+      },
+      ExpressionAttributeNames: {
+        '#NhsNumber': 'NhsNumber',
+      },
+      KeyConditionExpression: '#NhsNumber = :nhsNumber',
+    };
+    if (!includeDeletedRecord) {
+      params.FilterExpression = 'attribute_not_exists(DeletedAt)';
+    }
+    const command = new QueryCommand(params);
+
+    const response = await this.client.send(command);
+    const items = response?.Items;
+    if (!items) {
+      logError('Received an empty response from dynamodb during query');
+      return [];
+    }
+    return items;
+  }
+
+  async queryTableByConversationId(
+    inboundConversationId,
+    recordType = RecordType.ALL,
+    includeDeletedRecord = false
+  ) {
+    const params = {
+      TableName: this.tableName,
+      ExpressionAttributeNames: {
+        '#PrimaryKey': 'InboundConversationId',
+      },
+      ExpressionAttributeValues: {
+        ':InboundConversationId': inboundConversationId,
+      },
+      KeyConditionExpression: '#PrimaryKey = :InboundConversationId',
+    };
+    if (!includeDeletedRecord) {
+      params.FilterExpression = 'attribute_not_exists(DeletedAt)';
+    }
+
+    switch (recordType) {
+      case RecordType.ALL:
+        break;
+      case RecordType.CONVERSATION:
+      case RecordType.CORE:
+      case RecordType.FRAGMENT:
+        params.ExpressionAttributeNames['#sortKey'] = 'Layer';
+        params.ExpressionAttributeValues[':sortKey'] = recordType;
+        params.KeyConditionExpression += ' AND begins_with(#sortKey, :sortKey)';
+        break;
+      default:
+        logInfo(`Received unexpected queryType: ${recordType}. Will treat it as 'ALL'.`);
+    }
+
+    const command = new QueryCommand(params);
+
+    const response = await this.client.send(command);
+    const items = response?.Items;
+    if (!items) {
+      logError('Received an empty response from dynamodb during query');
+      return [];
+    }
+    return items;
+  }
+
+  async getItemByKey(inboundConversationId, inboundMessageId, recordType = RecordType.FRAGMENT) {
+    const expectedTypes = [RecordType.CORE, RecordType.FRAGMENT];
+
+    if (!expectedTypes.includes(recordType)) {
+      throw new Error('recordType has to be either Core or Fragment');
+    }
+    if (!inboundConversationId && !inboundMessageId) {
+      throw new Error('must be called with both conversationId and inboundMessageId');
```

Comment:
could you replace "conversationId" with "inboundConversationId" for clarification.

Reply:
Thanks, updated the error message